graphene_sqlalchemy/fields.py

```diff
@@ -94,18 +98,46 @@ def __init__(self, type, *args, **kwargs):
         super(SQLAlchemyConnectionField, self).__init__(type, *args, **kwargs)
 
 
-__connectionFactory = UnsortedSQLAlchemyConnectionField
+def default_connection_field_factory(relationship, registry):
+    model = relationship.mapper.entity
+    _type = registry.get_type_for_model(model)
+    return UnsortedSQLAlchemyConnectionField(_type._meta.connection)
+
+
+__connection_field_factory =  default_connection_field_factory
 
+def create_connection_field(relationship, registry):
+    return __connection_field_factory(relationship, registry)
 
-def createConnectionField(_type):
```

Comment:
Maybe you should also keep `createConnectionField` with a deprecated warning ?

Reply:
Fair point. This function is supposed to be private to the library but we can't guarantee that.